crates/libafl_intelpt/src/linux.rs

```diff
@@ -833,6 +834,31 @@ pub(crate) fn availability_in_linux() -> Result<(), String> {
     }
 }
 
+/// Check if the `MSR IA32_VMX_MISC` has bit 14 set, which indicates that Intel PT is
+/// available in VMX operations.
+pub(crate) fn availability_in_vmx() -> Result<(), String> {
+    let error_prefix = "Failed to check for PT availability in VM operations: ";
+    let mut msrs = fs::OpenOptions::new()
+        .read(true)
+        .open("/dev/cpu/0/msr")
+        .map_err(|e| {
+            format!(
+                "{error_prefix} Failed to open /dev/cpu/0/msr: {e} \
+                Make sure you have the `msr` kernel module loaded"
+            )
+        })?;
+    msrs.seek(SeekFrom::Start(0x485))
+        .map_err(|e| format!("{error_prefix} Failed to seek in /dev/cpu/0/msr: {e}"))?;
+    let mut buf = [0u8; 8];
+    msrs.read_exact(&mut buf)
+        .map_err(|e| format!("{error_prefix} Failed to read MSR 0x485: {e}"))?;
+    if buf[1] & 0b0100_0000 == 0 {
+        return Err("Intel PT is not available for VMX operations.".to_owned());
```

Comment:
Return a bool here, so error is "could not check" and then you return the actual check